pages/reference/OS/configuration.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Configuration
+excerpt: Instructions for editing and customizing config.json for modifying the behavior of {{ $names.os.lower }}
+---
+
+# Configuring {{ $names.os.lower }}
+
+The behavior of {{ $names.os.lower }} can be configured by editing the `config.json` file that resides in the boot partition of the device. This file accepts a [range of fields](#valid-fields) to modify the behavior of the host OS.
+
+__Note:__ The `config.json` file is different from the [`config.txt`](https://www.balena.io/docs/reference/OS/advanced/#configtxt) file also located in the boot partition, which is used by the Raspberry Pi to set device configuration options.
+
+## Using config.json to provision a device
+
+The `config.json` file is located in the root of the `resin-boot` partition, and you may edit this file by mounting a flashed SD card and editing the file directly. Editing this file will only work if you edit before the device's first boot. Once the device has been provisioned, edits to this file have no effect as the contents of the file are copied to the device boot partition at `/mnt/boot/config.json`.
+
+## Sample config.json
+
+The following example provides all available configuration options available in `config.json`. Full details about each value may be found in the [valid fields](#valid-fields) section.
+
+```json
+{
+  "hostname": "my-custom-hostname",
+  "persistentLogging": true,
+  "country": "GB",
+  "ntpServers": "ntp-wwv.nist.gov resinio.pool.ntp.org",
+  "dnsServers": "208.67.222.222 8.8.8.8",
+  "os": {
+    "network": {
+      "connectivity": {
+        "uri": "https://api.balena-cloud.com/connectivity-check",
+        "interval": "300",
+        "response": "optional value in the response"
+      },
+      "wifi": {
+        "randomMacAddressScan": false
+      }
+    },
+    "udevRules": {
+      "56": "ENV{ID_FS_LABEL_ENC}==\"resin-root*\", IMPORT{program}=\"resin_update_state_probe $devnode\", SYMLINK+=\"disk/by-state/$env{RESIN_UPDATE_STATE}\"",
+      "64": "ACTION!=\"add|change\", GOTO=\"modeswitch_rules_end\"\nKERNEL==\"ttyACM*\", ATTRS{idVendor}==\"1546\", ATTRS{idProduct}==\"1146\", TAG+=\"systemd\", ENV{SYSTEMD_WANTS}=\"u-blox-switch@'%E{DEVNAME}'.service\"\nLBEL=\"modeswitch_rules_end\"\n"
+    },
+    "sshKeys": [
+      "ssh-rsa AAAAB3Nza...M2JB balena@macbook-pro",
+      "ssh-rsa AAAAB3Nza...nFTQ balena@zenbook"
+    ]
+  }
+}
+```
+
+## Valid fields
```

Comment:
This section looks a lot like: https://github.com/balena-os/meta-balena#configjson
So there's a source of truth / sync issue: when someone merges a meta-balena PR that modifies the README, how does it get (automatically) propagated to the docs repo? We have an existing setup with scripts that copy md files from other repos - cli, supervisor, Node SDK... Should we setup something like that? Maybe the config.json docs could be moved out of the meta-balena README and into its own separate md file - also in the meta-balena repo?

Reply:
Yes, this is largely taken verbatim with a few fixes and that's a very good idea. I was thinking along those same lines by importing them as a partial. Let's complete this document in this repo for now (otherwise trying to coordinate reviews in two repos would be painful) and then once it is complete I'll move this section, to a new page in the meta-balena repo.

Reply:
This will be addressed in a followup PR. I've extracted this to a partial and from the PR will be generated from the meta-balena repo.